pkg/datastore/database.go

```diff
@@ -88,24 +88,22 @@ func (db *relationalDb) GetDBTransaction(ctx context.Context, tableName string,
 }
 
 func (db *relationalDb) getDBTransaction(ctx context.Context, tableName string, record Record) (tx *gorm.DB, err error) {
-	err, tenancyInfo := db.getTenantInfoFromCtx(ctx, tableName)
-	if err != nil {
+	var tenancyInfo TenancyInfo
+	if err, tenancyInfo = db.getTenantInfoFromCtx(ctx, tableName); err != nil {
 		return nil, err
 	}
 
-	err = db.ValidateTenancyScope(tenancyInfo, record, tableName)
-	if err != nil {
+	if err = db.ValidateTenancyScope(tenancyInfo, record, tableName); err != nil {
 		return nil, err
 	}
 
-	tx, err = db.ConfigureTxWithTenancyScope(tenancyInfo)
-	if err != nil {
+	if tx, err = db.ConfigureTxWithTenancyScope(tenancyInfo); err != nil {
```

Comment:
I think that since `ConfigureTxWithTenancyScope` is not interface's method and not exposed to the consumer, it should be made private (lowercase)